common/viperutil/config_util.go

```diff
@@ -24,14 +28,230 @@ import (
+func (c *ConfigParser) SetConfigFile(in string) {
+	if in != "" {
+		c.configFile = in
+	}
+}
+
+// Return the used configFile to fill config store
+func (c *ConfigParser) ConfigFileUsed() string { return c.configFile }
+
+// Set the env variable prefix
+// If "peer" is set here, all environment variable
+// searches shall be prefixed with "PEER_"
+func (c *ConfigParser) SetEnvPrefix(in string) {
+	if in != "" {
+		c.envPrefix = in
+	}
+}
+
+// Search for the existence of filename for all supported extensions
+func (c *ConfigParser) searchInPath(in string) (filename string) {
+	logger.Debugf("Searching for config in ", in)
+	for _, ext := range SupportedExts {
+		fullPath := filepath.Join(in, c.configName+"."+ext)
+		logger.Debugf("Checking for ", fullPath)
+		_, err := os.Stat(fullPath)
+		if err == nil {
+			logger.Debugf("Found: ", fullPath)
+			return fullPath
+		}
+	}
+	return ""
+}
+
+// Search for the configName in all configPaths
+func (c *ConfigParser) findConfigFile() string {
+
+	logger.Debugf("Searching for config in ", c.configPaths)
+
+	for _, cp := range c.configPaths {
+		file := c.searchInPath(cp)
+		if file != "" {
+			return file
+		}
+	}
+	return ""
+}
+
+// Get the valid and present config file
+func (c *ConfigParser) getConfigFile() string {
+	// if explicitly set, then use it
+	if c.configFile != "" {
+		return c.configFile
+	}
+
+	c.configFile = c.findConfigFile()
+	return c.configFile
+}
+
+func (c *ConfigParser) ReadInConfig() error {
+
+	cf := c.getConfigFile()
+	logger.Debugf("Attempting to read from config file : %s", cf)
+	file, err := ioutil.ReadFile(cf)
+	if err != nil {
+		logger.Errorf("Unable to read from config file : %s", cf)
+		return err
+	}
+
+	//c.config = make(map[string]interface{})
+
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(bytes.NewReader(file))
+	// read the configFile in config
+	return yaml.Unmarshal(buf.Bytes(), c.config)
+}
+
+// Identify the configFile from the provided configPaths
+// and read and unmarshall and store in config.
+func (c *ConfigParser) ReadConfig(in io.Reader) error {
+	//c.config = make(map[string]interface{})
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(in)
+	return yaml.Unmarshal(buf.Bytes(), c.config)
+}
+
+// Specifies the replacement string.
+// Used in searching env. variables.
+func (c *ConfigParser) SetEnvKeyReplacer(r *strings.Replacer) {
+	c.envKeyReplacer = r
+}
+
+// Set the environment variable search boolean
+func (c *ConfigParser) AutomaticEnv() {
+	c.automaticEnvApplied = true
+}
+
+// Get value for the key by searching
+// environment variables
+func (c *ConfigParser) GetFromEnv(key string) interface{} {
+	envKey := key
+	if c.envPrefix != "" {
+		envKey = strings.ToUpper(c.envPrefix + "_" + envKey)
+	}
+	if c.envKeyReplacer != nil {
+		envKey = c.envKeyReplacer.Replace(envKey)
+	}
+	return os.Getenv(envKey)
+}
+
+// Return the parsed config store
+func (c *ConfigParser) GetParsedConfig() map[string]interface{} {
+	return c.config
+}
```

Comment:
This function is only used by `EnhancedExactUnmarshal` and can probably be unexported.